### PR TITLE
just put impersonate info into jwe to make the cookie shorter

### PR DESCRIPTION
**Pull request by aiwantaozi**
Cookies are usually limited to 4096 bytes, when a user inputs the kubeconfig with certificate, the cookie may pretty long, so after user pass the kubernete access check, just put the impersonate info into the jwtToken.

### CODE REVIEW

**Comment by aiwantaozi** [comment]
1. The second request to create templateVersion is failed, when user don't select any image, no need to put dataVolumeTemplates in the template
2. When create templateVersion failed, now UI doesn't show the error, need to show error when the request failed.

image
```
Response:
{"type":"error","links":{},"code":"Invalid","message":"VirtualMachineTemplateVersion.harvester.cattle.io \"without-image-dbdls\" is invalid: spec.vm.dataVolumeTemplates.spec.source.http.url: Required value","status":422,"type":"error"}
```
```
Request Body:
{"apiVersion":"harvester.cattle.io/v1alpha1","kind":"harvester.cattle.io.virtualmachinetemplateversion","type":"harvester.cattle.io.virtualmachinetemplateversion","spec":{"templateId":"harvester-system/without-image","keyPairIds":[],"vm":{"dataVolumeTemplates":[{"apiVersion":"cdi.kubevirt.io/v1alpha1","kind":"DataVolume","metadata":{"name":"without-image-disk-0-ljyru","annotations":{}},"spec":{"pvc":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"10Gi"}},"volumeMode":"Filesystem","storageClassName":"hostpath"},"source":{"http":{}}}}],"template":{"spec":{"domain":{"cpu":{"cores":1,"sockets":1,"threads":1},"devices":{"interfaces":[{"masquerade":{},"model":"virtio","name":"default"}],"disks":[{"disk":{"bus":"virtio"},"name":"disk-0"},{"name":"cloudinitdisk","disk":{"bus":"virtio"}}]},"resources":{"requests":{"memory":"1Gi"}}},"networks":[{"pod":{},"name":"default"}],"volumes":[{"name":"disk-0","dataVolume":{"name":"without-image-disk-0-ljyru"}},{"name":"cloudinitdisk","cloudInitNoCloud":{"userData":"#cloud-config\nssh_authorized_keys: []\n"}}]},"metadata":{"labels":{"harvester.cattle.io/creator":"harvester","harvester.cattle.io/vmName":"without-image"}}},"running":true}}}
```

File: pkg/api/auth/token_test.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		authInfo, err := buildAuthInfo(tc.given.token, tc.given.kubeconfig)
+		assert.Nil(t, err, "case %q", tc.name)
+		if err != nil {
+			continue
+		}
```
Comment (gitlawr) [inline pkg/api/auth/token_test.go]: err is asserted to be nil so we don't need this check

**Review by guangbochen — COMMENTED** [review]
will this require any front-end changes?

File: pkg/api/auth/token_test.go (hunk 1 above)
Comment (aiwantaozi) [reply pkg/api/auth/token_test.go:83]: When the error isn't empty, after assert.Nil it still continue to the next step, no need to run the below test when the first step is failed.

**Comment by aiwantaozi** [comment]
UI no need to change

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]

**Pull request merged**